src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -352,7 +352,11 @@ public function copy($path, $target)
     public function link($target, $link)
     {
         if (! windows_os()) {
-            return symlink($target, $link);
+            if(function_exists('symlink')) {
+                return symlink($target, $link);
+            }else{
+                return exec("ln -s ".escapeshellarg($target).' '.escapeshellarg($link));
+            }
         }
```

Comment:
This can be simplified
```suggestion
        if (! windows_os() && function_exists('symlink')) {
            return symlink($target, $link);
        }
        
        if (! windows_os()) {
                return exec("ln -s ".escapeshellarg($target).' '.escapeshellarg($link));
        }
```

Reply:
Maybe even better, the method could be rewritten to this:
```php
    public function link($target, $link)
    {
        if (windows_os()) {
            $mode = $this->isDirectory($target) ? 'J' : 'H';
            exec("mklink /{$mode} ".escapeshellarg($link).' '.escapeshellarg($target));
        }

        if(function_exists('symlink')) {
            return symlink($target, $link);
        }

        return exec("ln -s ".escapeshellarg($target).' '.escapeshellarg($link));
    }
```